conversions/hexadecimal_to_decimal

```diff
@@ -0,0 +1,32 @@
+def hex_to_decimal(hex_string: str) -> int:
+    """
+    converts a given hexadecimal value to its decimal value
+    >>> hex_to_decimal("a")
+    10
+    >>> hex_to_decimal("")
+    ValueError: Empty string value was passed to the function
+    >>> hex_to_decimal("-12m")
+    ValueError: Non-hexadecimal value was passed to the function
+    >>> hex_to_decimal("12f")
+    303
+    >>> hex_to_decimal("   12f   ")
+    303
+    """
```

Comment:
```suggestion
    >>> hex_to_decimal("FfFf")
    65535
    """
```
Let's make sure we can handle longer numbers and mixed case.

Reply:
Do we want to deal with negative numbers?

Reply:
The code raises an error if it finds negative sign as it's not a valid character in our table

Reply:
We should be able to do as well as Python's builtin `int()` function...
$ python3
```
>>> int(-0xFf)
-255
```